pkg/kubectl/bind-apiservice/plugin/servicebindings.go

```diff
@@ -139,13 +141,156 @@ func (b *BindAPIServiceOptions) createAPIServiceBindings(ctx context.Context, co
 	return bindings, nil
 }
 
+func printPermissionClaim(w io.Writer, p kubebindv1alpha1.PermissionClaim) error {
+	var b bytes.Buffer
+
+	var groupResource string
+	if p.GroupResource.Group != "" {
+		groupResource = fmt.Sprintf("%s objects (apiVersion: \"%s/%s\")", p.GroupResource.Resource, p.GroupResource.Group, p.Version)
+	} else {
+		groupResource = fmt.Sprintf("%s objects (apiVersion: \"%s\")", p.GroupResource.Resource, p.Version)
+	}
+
+	if err := writeFirstLines(&b, groupResource, p); err != nil {
+		return err
+	}
+
+	if err := writeOnConflict(&b, p); err != nil {
+		return err
+	}
+
+	if err := writeUpdateClause(&b, p); err != nil {
+		return err
+	}
+
+	if err := writeRequiredAndAcceptance(&b, p.Required); err != nil {
+		return err
+	}
+
+	_, err := fmt.Fprint(w, b.String())
+	return err
+}
+
+func writeFirstLines(b *bytes.Buffer, groupResource string, claim kubebindv1alpha1.PermissionClaim) error {
+	var err error
+
+	donate := false
+	if claim.Create != nil {
+		donate = claim.Create.Donate
+	}
+	adopt := claim.Adopt
+
+	name := ""
+	var owner kubebindv1alpha1.Owner
+	if (claim.Selector != kubebindv1alpha1.ResourceSelector{}) {
+		name = claim.Selector.Name
+		owner = claim.Selector.Owner
+	}
+	var postamble string
+	switch {
+	case !donate && !adopt:
+		groupResource = "read " + groupResource
+	case donate && !adopt:
+		groupResource = "create " + groupResource
+	case !donate && adopt:
+		groupResource = "have ownership of " + groupResource
+		postamble = " and assign the ownership to you"
+	case donate && adopt:
+		groupResource = "create" + groupResource
+		postamble = " and wants to keep the ownership"
+	}
+
+	_, err = fmt.Fprintf(b, "The provider wants to %s on your cluster%s.", groupResource, postamble) //xxx
```

Comment:
What does xxx mean?

Reply:
Looks like i missed to remove this after testing. I removed it in b9eac5dffe6fa930fd8ef4f2db5ead3c8cb1918f.